core/templates/dev/head/forms/formBuilder.js

```diff
@@ -598,9 +603,22 @@ oppia.factory('rteHelperService', [
       // Returns a DOM node.
       createRteElement: function(componentDefn, customizationArgsDict) {
         var el = $('<img/>');
-        el.attr('src', componentDefn.iconDataUrl);
+        if (componentDefn.previewUrlTemplate) {
+          el.attr('src',
+            $interpolate(
+              componentDefn.previewUrlTemplate)(
+                angular.extend(customizationArgsDict,
+                  {
+                    explorationId: explorationContextService.getExplorationId()
```

Comment:
(Btw, for testing purposes, you should be able to just easily create an RTE component in some other page, the same way as we do it in e.g. state_editor_content.html using the schema.)